source/slang/slang-ir-redundancy-removal.cpp

```diff
@@ -177,11 +178,29 @@ void removeAvailableInDownstreamModuleDecorations(CodeGenTarget target, IRModule
                     // Gut the function definition, turning it into a declaration
                     for (auto inst : funcInst->getChildren())
                     {
+                        // For HLSL, we can delete the whole block
+                        // but for SPIR-V we have to keep the block with only the
+                        // parameters inside, no other instructions.
                         if (inst->getOp() == kIROp_Block)
                         {
-                            toRemove.add(inst);
+                            if (target == CodeGenTarget::HLSL)
+                            {
+                                toRemove.add(inst);
+                            }
+                            else if (target == CodeGenTarget::SPIRV)
+                            {
+                                for (auto blockInst : inst->getChildren())
```

Comment:
You cannot modify the children while iterating it with getChildren.
need to call `inst->getModifiableChildren()`.

See https://github.com/shader-slang/slang/blob/master/docs/design/ir.md#ir-global-and-hoistable-value-deduplication

Reply:
Isn't it acceptable that I'm just adding the block instructions to a list of things, and then I remove them all at the end?

Reply:
Ah you are right that works too.